firedrake/interpolation.py

```diff
@@ -85,6 +87,35 @@ class Interpolator(object):
 
     """
     def __init__(self, expr, V, subset=None, freeze_expr=False, access=op2.WRITE, bcs=None):
+
+        # NOTE: below redoes a bunch of checks which happen in
+        # make_interpolator. I could perhaps try and catch a custom exeception
+        # in the existing code and then check if I'm doing a VOM onto VOM
+        arguments = extract_arguments(expr)
+        if len(arguments) == 0:
+            target_mesh = V.ufl_domain()
+            source_mesh = extract_unique_domain(expr) or target_mesh
+        elif len(arguments) == 1:
+            if isinstance(V, firedrake.Function):
+                raise ValueError(
+                    "Cannot interpolate an expression with an argument into a Function"
+                )
+            argfs = arguments[0].function_space()
+            target_mesh = V.ufl_domain()
+            source_mesh = argfs.mesh()
+        else:
+            raise ValueError(
+                "Cannot interpolate an expression with %d arguments" % len(arguments)
+            )
+        if (
+            target_mesh is not source_mesh
+            and isinstance(target_mesh.topology, firedrake.mesh.VertexOnlyMeshTopology)
+            and isinstance(source_mesh.topology, firedrake.mesh.VertexOnlyMeshTopology)
+        ):
+            self.vom_onto_vom = True
+        else:
+            self.vom_onto_vom = False
```

Comment:
The pattern:
```
if condition:
    foo = True
else:
    foo = False
```
is generally better written as:
```
foo = condition
```

Also `vom_onto_vom` is an excellent variable name.